From b32caff43b040be34c206962402e881e4235c5a9 Mon Sep 17 00:00:00 2001
From: Andy Green <andy@openmoko.com>
Date: Sun, 13 Apr 2008 07:25:51 +0100
Subject: [PATCH] fix-charging-deassert-host-power-1a-detect.patch

We don't take care to stop driving generated USB host power even
when we have a 1A charger connected on the same pins.

Signed-off-by: Andy Green <andy@openmoko.com>
---
 drivers/i2c/chips/pcf50633.c |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/drivers/i2c/chips/pcf50633.c b/drivers/i2c/chips/pcf50633.c
index 91b9ac3..c14fad0 100644
--- a/drivers/i2c/chips/pcf50633.c
+++ b/drivers/i2c/chips/pcf50633.c
@@ -523,6 +523,15 @@ static void configure_pmu_for_charger(struct pcf50633_data *pcf,
 		break;
 	case CHARGER_TYPE_1A:
 		__reg_write(pcf, PCF50633_REG_MBCC7, PCF50633_MBCC7_USB_1000mA);
+		/*
+		 * stop GPO / EN_HOSTUSB power driving out on the same
+		 * USB power pins we have a 1A charger on right now!
+		 */
+		dev_info(&pcf->client.dev, "Charger -> CHARGER_TYPE_1A\n");
+		__reg_write(pcf, PCF50633_GPO - PCF50633_GPIO1 +
+				 PCF50633_REG_GPIO1CFG,
+				 __reg_read(pcf, PCF50633_GPO - PCF50633_GPIO1 +
+						 PCF50633_REG_GPIO1CFG) & 0xf0);
 		break;
 	}
 }
-- 
1.5.6.5

